File: backend/domain/src/main/kotlin/pt/ira/occurrence/OccurrenceType.kt
package pt.ira.occurrence

/**
 * Define os níveis de importância de uma ocorrência.
 *
 * Estes níveis podem ser utilizados para priorização, alertas e workflows
 * diferenciados no sistema.
 *
 * - [NORMAL]: Ocorrência de prioridade padrão, sem urgência.
 * - [URGENT]: Ocorrência que requer atenção rápida.
 * - [CRITICAL]: Ocorrência crítica que exige intervenção imediata.
 */
enum class OccurrenceType {
    NORMAL,
    URGENT,
    CRITICAL,
}

File: frontend/commons/models/occurrence/OccurrenceType.ts
/**
 * Nível de importância de uma ocorrência.
 * - NORMAL: Baixa prioridade, sem urgência.
 * - URGENT: Requer atenção rápida.
 * - CRITICAL: Máxima prioridade, requer resposta imediata.
 */
export enum OccurrenceType {
  NORMAL = "NORMAL",
  URGENT = "URGENT",
  CRITICAL = "CRITICAL",
}
